jade-webkit.el

```diff
@@ -235,15 +247,69 @@ same url."
                      (funcall callback message)))
        (t (pcase method
             ("Console.messageAdded" (jade-webkit--handle-console-message message))
+            ("Debugger.scriptParsed" (jade-webkit--handle-script-parsed message))
             ("Debugger.paused" (jade-webkit--handle-debugger-paused message))
             ("Debugger.resumed" (jade-webkit--handle-debugger-resumed message))))))))
 
 (defun jade-webkit--handle-console-message (message)
+  "Emit console MESSAGE to repl buffer."
   (let* ((msg (map-nested-elt message '(params message)))
-         (parameters (map-elt msg 'parameters)))
+         (parameters (map-elt msg 'parameters))
+         (url (map-elt msg 'url))
+         (line (map-elt msg 'line))
+         (column (map-elt msg 'column))
+         (position (jade-webkit--find-source-map-position url line column)))
+    (when position
+      (setf (map-elt msg 'url) (plist-get position :source))
+      (setf (map-elt msg 'line) (plist-get position :line))
+      (setf (map-elt msg 'colum) (plist-get position :column)))
     (setf (map-elt msg 'parameters) (seq-map #'jade-webkit--value parameters))
     (jade-repl-emit-console-message msg)))
 
+(defun jade-webkit--find-source-map-position (url line column)
+  "Find sourcemap position by corresponding URL, LINE and COLUMN."
+  (let ((map (map-elt jade-webkit-source-maps url)))
+    (when map
+      (sourcemap-original-position-for (map-elt map 'source-map)
+                                       :line line
+                                       ;; hack for sourcemap
+                                       :column (1- column)))))
```

Comment:
Well it won't work without it. I've checked actual data and found that I have to add this decrement to make sourcemaps work.